pennylane/devices/qubit/simulate.py

```diff
@@ -83,6 +99,43 @@ def expand_state_over_wires(state, state_wires, all_wires, is_state_batched):
     return qml.math.transpose(state, desired_axes)
 
 
+def _postselection_postprocess(state, is_state_batched, shots):
+    """Update state after projector is applied."""
+    if is_state_batched:
+        raise ValueError(
+            "Cannot postselect on circuits with broadcasting. Use the "
+            "qml.transforms.broadcast_expand transform to split a broadcasted "
+            "tape into multiple non-broadcasted tapes before executing if "
+            "postselection is used."
+        )
+
+    # The floor function is being used here so that a norm very close to zero becomes exactly
+    # equal to zero so that the state can become invalid. This way, execution can continue, and
+    # bad postselection gives results that are invalid rather than results that look valid but
+    # are incorrect.
+    norm = qml.math.floor(qml.math.real(qml.math.norm(state)) * 1e15) * 1e-15
+
+    if shots:
+        # Clip the number of shots using a binomial distribution using the probability of
+        # measuring the postselected state.
+        postselected_shots = (
+            (
+                np.random.binomial(shots.total_shots, float(norm))
+                if not shots.has_partitioned_shots
+                else [np.random.binomial(s, float(norm)) for s in shots]
+            )
```

Comment:
nit, but this works just as well
```suggestion
            np.random.binomial(list(shots), float(norm)).tolist()
```

Reply:
That's what I had before. I just changed it for the sake of coverage, but I guess if we do this I don't need to check in `_FlexShots` if `shots` is an integer or a sequence.

Reply:
Fixed.